pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -272,11 +271,8 @@ func (vc *VirtualCenter) Connect(ctx context.Context) error {
 func (vc *VirtualCenter) connect(ctx context.Context, requestNewSession bool) error {
 	log := logger.GetLogger(ctx)
 
-	if _, ok := clientMutex[vc.Config.Host]; !ok {
-		clientMutex[vc.Config.Host] = &sync.Mutex{}
-	}
-	clientMutex[vc.Config.Host].Lock()
-	defer clientMutex[vc.Config.Host].Unlock()
+	vc.ClientMutex.Lock()
```

Comment:
calling `vc.ClientMutex.Lock()` and then immediately calling `vc.ClientMutex.Unlock()` has no effect.

Reply:
Forgot to add defer before Unlock. Just updated it.